retrievalmarket/client/client.go

```diff
@@ -549,14 +549,16 @@ func (c *Client) retrieveContentFromPeerWithProgressCallback(
 	chanid = newchid
 	chanidLk.Unlock()
 
-	defer func() { _ = c.dataTransfer.CloseDataTransferChannel(ctx, chanid) }()
```

Comment:
We certainly should not have the `defer func() { _ = c.dataTransfer.CloseDataTransferChannel(ctx, chanid) }()` here as it is called even on successful/completed transfers and we don't want that.